design-proposals/003-credentials.md

```diff
@@ -0,0 +1,226 @@
+# Credential support
+
+* Issue: https://github.com/syndesisio/syndesis-rest/issues/386
+* Sprint: 13
+* Affected Repos:
+  - syndesis-rest
+  - syndesis-ui
+  
+## Background
+
+High level goals are:
+
+ * Help users provide and manage credentials
+ * Persist credentials in a secure manner
+ * Apply credentials to connections with minimum impact on the existing implementation
+
+## User Story
+
+As a citizen user I would like to connect to 3rd party service that require me to authenticate using credentials. Credentials that I provide or that are received during the authorization with the 3rd party service would need to be stored in Syndesis so I can manage them. For example when connecting to Salesforce the I grant Syndesis rights to perform actions on my behalf by, and in order to do that I would like to choose "connect to Salesforce" in Syndesis authenticate against Salesforce to authorize that access and not be prompted again for authorization. Syndesis in turn would apply the credentials received in the authorization process with Salesforce to my Salesforce connection, and perform any management needed, such as refreshing the the received credentials. I would like to have the ability to manage credentials that are stored in Syndesis.
```

Comment:
As said before, I don't think that you create another connection with the same authorization. If you use a different connection for the same connector you do this very likely for a different identity, too (otherwise just reuse the existing connection).